models/dvmdostem/R/model2netcdf.dvmdostem.R

```diff
@@ -6,14 +6,112 @@
+        } else {
+          PEcAn.logger::logger.error(paste0("ERROR!: ", k, " is not a monthly or yearly variable!"))
+          stop()
+        }
+
+        # Get the data
+        vardata <- ncdf4::ncvar_get(ncin_tr_y, k)
+
+        # Look up the units in dvmdostem world
+        original_units <- ncdf4::ncatt_get(ncin_tr_y, k, "units")
+        original_units <- as.character(unlist(original_units)[2]) # How to avoid hard coded index??
+
+        # This is a temporary hack, till dvm-dos-tem issue is resolved...
+        # See issue #336 (https://github.com/ua-snap/dvm-dos-tem/issues/336)
+        original_units <- gsub("time", "year", original_units)
+        original_units <- gsub("gC", "g C", original_units)
+
+        # Convert the data
+        vardata_new <- PEcAn.utils::misc.convert(vardata, original_units, vmap_reverse[[j]][["newunits"]])
+
+        # Coerce the data into the right shape (y, x, time).
+        # With a single pixel run, the Y and X dimensions are lost when
+        # reading from the file with ncdf4::ncvar_get, and the subsequent 
+        # ncdf4::ncvar_put call fails. So here we make sure that the
+        # vardata_new data is a 3D structure:
+        dim_lengths <- sapply(ncin_tr_y$var[[1]]$dim, function(x) x$len)
+        vardata_new <- array(vardata_new, dim = dim_lengths)
+
+        dim.order <- sapply(ncin_tr_y$var[[k]]$dim, function(x) x$name)
+        starts <-c(y = px_Y, x = px_X, time = 1)
+
+
+        if (TRUE %in% sapply(monthly_dvmdostem_outputs, function(x) grepl(paste0("^",k,"_"), x))) {
+          # The current variable (j) is a monthly output
+          counts <- c(y=1, x=1, time=12)
+          startidx <- ((i-1)*12)+1
+          endidx <- i*12
+          newVector <- cbind(newVector, vardata_new[px_X, px_Y,startidx:endidx])
+        } else if (TRUE %in% sapply(yearly_dvmdostem_outputs, function(x) grepl(paste0("^",k,"_"), x))) {
+          # The current variable (k) is a yearly output
+          counts <- c(y=1, x=1, time=1)
+          newVector <- cbind(newVector, vardata_new[px_X, px_Y, i])
+        } else {
+          PEcAn.logger::logger.error(paste0("ERROR!: ", k, " is not a monthly or yearly variable!"))
+          stop()
+        }
+
+      }
+
+      # Maybe we will support more operations in the future besides sum...
+      newVector <- apply(newVector, 1, sum)
+
+      # Add new data to netcdf file...
+      ncdf4::ncvar_put(ncout, j, newVector, start = starts[dim.order], count = counts[dim.order])
+    }
+    ncdf4::nc_close(ncout)
+  }
+}
+
 ##-------------------------------------------------------------------------------------------------#
 ##' @name model2netcdf.dvmdostem
 ##' @title Code to convert dvmdostem netcdf output into into CF standard
 ##' 
 ##' @param outdir Location of dvmdostem model output
 ##' @param runstart ??
 ##' @param runend ??
-##' 
+##' @param pecan_requested_vars a space separated string with names of the PEcAn variables to output.
```

Comment:
can you provide an example of you what want here?  I feel like space separated is a bad way to go and better to have a comma list unless there is a good reason?

Reply:
@serbinsh, not sure why I chose space delimited here, but you are right probably a bad idea. I will change to comma.

Reply:
@serbinsh, ok, just pushed a few more changes that should change to a comma separated implementation.